### DISCUSSION TITLE
Why can't the reader that xmlLinq gives me serialize binary content?

### DISCUSSION THREAD

**Discussion by zms9110750** [dotnet/runtime#73944] [Q&A]
```csharp
var element = new XElement("xml");

var read1 = element.CreateReader();
var read2 = XmlReader.Create(read1, new XmlReaderSettings());

Console.WriteLine(read1.CanReadBinaryContent);//False
Console.WriteLine(read2.CanReadBinaryContent);//True

Console.WriteLine(read1);//System.Xml.Linq.XNodeReader
Console.WriteLine(read2);//System.Xml.XmlCharCheckingReader
```
In this case, why didn't you give me an xmlcharcheckingreader directly at the beginning?

>All Microsoft .NET Framework implementations of the XmlReader class return true for this property.

This makes me very puzzled

**Comment by huoyaoyuan** [comment]
> All Microsoft .NET Framework implementations of the XmlReader class return true for this property.

Which version are you using? Note that .NET 5+ isn't .NET Framework, they are .NET Core.

We may remove this from .NET Core documentation.

**Reply by zms9110750** [reply]
Neither version works. I reproduce this example in net6. I found the problem in unity2021. He uses fromwork 4.7.X
